src/handle_init_contract.c

```diff
@@ -35,17 +23,24 @@ void handle_init_contract(void *parameters) {
     // Initialize the context (to 0).
     memset(context, 0, sizeof(*context));
 
-    uint32_t selector = U4BE(msg->selector, 0);
-    if (find_selector(selector, FIGMENT_SELECTORS, NUM_SELECTORS, &context->selectorIndex)) {
+    size_t index;
+    if (!find_selector(U4BE(msg->selector, 0), SELECTORS, SELECTOR_COUNT, &index)) {
+        PRINTF("Error: selector not found!\n");
```

Comment:
Are printf's ok to have in here, or do we need to write the message somewhere else?

Reply:
This part comes from the boilerplate plugin. I think it's the recommended way of reporting such errors.